docs/concepts/package-types.md

```diff
@@ -0,0 +1,225 @@
+  [`modular add`](../commands/add.md) command. Modular templates are not
+  guaranteed to be compatible with CRA templates.
+- Source files are loaded with the more performant
+  [`esbuild-loader`](https://github.com/privatenumber/esbuild-loader) in the
+  Webpack configuration, and Babel plugins are not supported.
+- [Local proxies](https://create-react-app.dev/docs/proxying-api-requests-in-development/)
+  are supported, but [esbuild mode](../configuration.md) doesn't support the
+  `package.json` `proxy` field. The more flexible
+  [manual proxy configration](https://create-react-app.dev/docs/proxying-api-requests-in-development/#configuring-the-proxy-manually)
+  is supported in both Webpack and esbuild mode.
+
+Apps can be built using [`modular build`](../commands/build.md): the resulting
+output is an optimized site that can be served statically. All code (files in
+`src` plus external dependencies required in the code) are bundled in a single
+blob of code that can be split in different files.
+
+Apps can be started using [`modular start`](../commands/start.md): a developer
+server runs on port 3000, serving the app with an additional runtime layer that
+provides developer experience functionalities like hot reloading and on-screen
+error overlay.
+
+## ESM View
+
+Modular `esm-view`s are similar to `app`s with some essential differences: they
+are compiled as [ES
+Modules(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules)]
+and all their external `import`s are rewritten to point to an
+[ESM CDN](../esm-views/esm-cdn.md). Like apps, they are be built with the same
+Webpack or esbuild configuration and support the same functionalities. Unlike
+apps, they don't allow the user to specify a custom `index.html`, they don't
+include a `public` folder in the build and they expect their entrypoint
+(`src/index.tsx`) to not `render` to the DOM, but to export something, typically
+a React Component.
+
+Esm Views are Modular's type of choice to implement the
+[micro-frontend](../concepts/microfrontends.md) pattern. They typically expect
+their entrypoint exports to be
+[dynamically `import`ed at runtime](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import)
+by some other micro-frontend, but can be also
+[served standalone](../esm-views/how-to-build.md) exactly as an `app`, provided
+that they default export a React component that can be rendered to the DOM.
+
+Since the micro-frontend pattern allows different teams to build and serve their
+applications independently and compose them at runtime (parallelizing otherwise
+expensive build operations), third-party dependencies need to be de-duplicated
+across micro-frontends; for this reason, external dependencies are not bundled,
+but they are rewritten to a [configurable ESM CDN](../esm-views/esm-cdn.md) that
+serves them as ES modules on the fly.
+
+Esm Views can be built using [`modular build`](../commands/build.md): the
```

Comment:
Following my comment about separate pages for each package type, I think each page should have headings for each of the [appropriate?] Modular commands (always in the same order), that explain the detail here and include a code snippet for e.g. `modular build my-view`, `modular add --type esm-view` etc. Don't worry about the duplication between pages; this is on purpose: some users will only view one page, and those who view several will quickly learn about the common interface that Modular exposes.

Optionally, you could link the answers in the table above to each specific section on the page to provide additional detail, if desired, instead of using longer sentences in the table.

Reply:
- The problem with `add` is that we have `--unstable-type`, derived from CRA, that we don't really want to show on the docs. The preferred mode of adding a page is the interactive prompt.
- The problem with `build` is that the command will always be the same.

But I like the idea of having structured information: I can do that for entry-points and to explain what happens when things are built or started, for example.